homeassistant/components/binary_sensor/isy994.py

```diff
@@ -49,22 +83,273 @@ def setup_platform(hass, config: ConfigType,
 
 
 class ISYBinarySensorDevice(isy.ISYDevice, BinarySensorDevice):
-    """Representation of an ISY994 binary sensor device."""
+    """Representation of an ISY994 binary sensor device.
+
+    Often times, a single device is represented by multiple nodes in the ISY,
+    allowing for different nuances in how those devices report their on and
+    off events. This class turns those multiple nodes in to a single Hass
+    entity and handles both ways that ISY binary sensors can work.
+    """
 
     def __init__(self, node) -> None:
         """Initialize the ISY994 binary sensor device."""
-        isy.ISYDevice.__init__(self, node)
+        super().__init__(node)
+        self._negative_node = None
+        self._heartbeat_device = None
+        self._device_class_from_type = self._detect_device_type()
+        # pylint: disable=protected-access
+        if self._node.status._val == -1*float('inf'):
+            self._computed_state = None
+        else:
+            self._computed_state = bool(self._node.status._val)
+
+    @asyncio.coroutine
+    def async_added_to_hass(self) -> None:
+        """Subscribe to the node and subnode event emitters."""
+        yield from super().async_added_to_hass()
+
+        self._node.controlEvents.subscribe(self._positive_node_control_handler)
+
+        try:
+            self._negative_node.controlEvents.subscribe(
+                self._negative_node_control_handler)
+        except AttributeError:
```

Comment:
Why not check `if self._negative_node is not None` instead?

Reply:
I went with the Python pattern of "ask for forgiveness" for no other reason than it seems like the Pythonic way. Happy to change it, and I'd love to hear if you have thoughts about when `except` should be used vs checking for None.

Reply:
If there's a simple check that is safe and always works, I go with an `if`. If there are multiple checks needed or if those checks don't always work, or if cases are unpredictable, I go with `try... except`. But it all depends... :smile:
Here the case looks contained to me since we initialize the attribute to `None`.

Reply:
Got it! Since we control that property in a closed environment, we should just check for values that we know it can be. I just pushed the change for these comments